specifications/api/swagger.yaml

```
@@ -3323,14 +3345,71 @@ components:
         - title: outputs without amounts
           <<: *transactionNoAmountOutputs
 
+    ApiAssetQuantity: &ApiAssetQuantity
+      type: object
+      required:
+        - quantity
+        - unit
+      properties:
+        quantity:
+          type: integer
+          example: 14
+        unit:
+          type: string
+          enum:
+            - assets
+
+    ApiMintData: &ApiMintData
+      type: object
+      required:
+        - receiving_address
+        - amount
+      properties:
+        receiving_address: *addressId
+        amount: *ApiAssetQuantity
+
+    ApiBurnData: &ApiBurnData
+      allOf:
+        - *ApiAssetQuantity
+
+    ApiMintBurnOperation: &ApiMintBurnOperation
+      type: object
+      oneOf:
+        - title: "mint"
+          properties:
+            mint: *ApiMintData
+        - title: "burn"
+          properties:
+            burn: *ApiBurnData
+
+    ApiMintBurnData: &ApiMintBurnData
+      type: object
+      required:
+        - asset_name
+        - operation
+        - policy_script_template
+      properties:
+        policy_script_template: *ScriptTemplateValue
+        verification_key_index:
+          allOf:
+            - *derivationSegment
+            - type: string
+              default: 0
+        asset_name: *assetName
+        operation: *ApiMintBurnOperation
+
     ApiConstructTransactionData: &ApiConstructTransactionData
       description: At least one field needs to be choosen
       type: object
       properties:
         payments: *ApiPaymentDestination
         withdrawal: *transactionWithdrawalRequestSelf
         metadata: *transactionMetadata
-        mint: *transactionMint
+        minted_burned:
```

```suggestion
        mint_burn:
```
I think `mint_burn` would be more consistent with other parameter names in this payload (which don't use past tense), like `payments` or `withdrawal`.

---

I agree with @piotr-iohk here, as `minted`/`burned` seems to imply the past tense, but this is for a transaction that has not yet been constructed. 😄 

For this reason, I think it's clearer to use `mint` and `burn` here.